src/Analyzers/CSharp/Tests/UseParameterNullChecking/UseParameterNullCheckingTests.cs

```diff
@@ -0,0 +1,981 @@
+        {
+            throw new ArgumentNullException(nameof(s));
+        }|]
+    }
+
+    void M2(string s)
+    {
+        [|if (" + className + @".ReferenceEquals(null, s))
+        {
+            throw new ArgumentNullException(nameof(s));
+        }|]
+    }
+}",
+                FixedCode = @"
+using System;
+
+class C
+{
+    private readonly string s;
+    void M1(string s!!)
+    {
+    }
+
+    void M2(string s!!)
+    {
+    }
+}",
+                LanguageVersion = LanguageVersionExtensions.CSharpNext
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseIsNullCheck)]
+        public async Task TestNotCustomReferenceEqualsCheck()
+        {
+            var testCode = @"
+using System;
+
+class C
+{
+    private readonly string s;
+    void M(string s)
+    {
+        if (ReferenceEquals(s, null))
+        {
+            throw new ArgumentNullException(nameof(s));
+        }
+    }
+
+    bool ReferenceEquals(object o1, object o2) => false;
+}";
+            await new VerifyCS.Test()
+            {
+                TestCode = testCode,
+                FixedCode = testCode,
+                LanguageVersion = LanguageVersionExtensions.CSharpNext
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseIsNullCheck)]
+        public async Task TestNotEqualitySwapped()
+        {
+            var testCode = @"using System;
+
+class C
+{
+    void M(string s)
+    {
+        if (null != (object)s)
+            throw new ArgumentNullException(nameof(s));
+    }
+}";
+            await new VerifyCS.Test()
+            {
+                TestCode = testCode,
+                FixedCode = testCode,
+                LanguageVersion = LanguageVersionExtensions.CSharpNext
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseIsNullCheck)]
+        public async Task TestMissingPreCSharp11()
+        {
+            var testCode = @"
+using System;
+
+class C
+{
+    void M(string s)
+    {
+        if ((object)s == null)
+            throw new ArgumentNullException(nameof(s));
+    }
+}";
+            await new VerifyCS.Test()
+            {
+                TestCode = testCode,
+                FixedCode = testCode,
+                LanguageVersion = LanguageVersion.CSharp10
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseIsNullCheck)]
+        public async Task TestOnlyForObjectCast()
+        {
+            var testCode = @"using System;
+
+class C
+{
+    void M(string s)
+    {
+        if ((string)s == null)
+            throw new ArgumentNullException(nameof(s));
+    }
+}";
+            await new VerifyCS.Test()
+            {
+                TestCode = testCode,
+                FixedCode = testCode,
+                LanguageVersion = LanguageVersionExtensions.CSharpNext
+            }.RunAsync();
```

Comment:
i suppose this is ok.  but it is somewhat odd since this would be totally safe to change to `!!`.  but it does seem rare.  we can wait to hear feedback.

Reply:
This imitates the behavior of "use 'is null' instead of '== null'".